chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -830,3 +831,34 @@ func (b *BitcoindNotifier) RegisterBlockEpochNtfn(bestBlock *chainntnfs.BlockEpo
 		}, nil
 	}
 }
+
+func (b *BitcoindNotifier) catchUpOnBlocks(bestBlock *chainntnfs.BlockEpoch) error {
+	if bestBlock == nil {
+		return nil
+	}
+
+	_, currHeight, err := b.chainConn.GetBestBlock()
```

Comment:
> Should this maybe use the best block contained within the `BitcoindNotifier` struct itself?

I believe it's correct to use the height of the best block contained within the `BitcoindNotifier` as the *starting point* for dispatching historical notifications, not the ending point as this comment suggests.
Reasoning: `catchUpOnMissedBlocks` is only called when a block has connected that is 1+ blocks ahead of `BitcoindNotifier`'s best block, for the purpose of catching up clients to the height of the newly connected block

> It might be the case that GetBestBlock returns a new block that the notifier isn't aware of yet, so the registered client would end up receiving two notifications for the same block.

That's a good point. In that case, I think it would be best to edit `catchUpOnMissedBlocks` as follows:

1) It takes in the block that just connected, rather than the `BitcoindNotifier`'s best block
2) It uses the height  of the block passed in from (1) as the condition for when to stop sending past notifications, rather than the height of what `GetBestBlock` returns. 

This should prevent duplicate notifications in case new blocks were received on-chain that haven't connected to the notifier yet

TL;DR true, will fix